build/codegen/codegen.sh

```diff
@@ -14,29 +14,48 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GROUP_VERSIONS="ceph.rook.io:v1"
+set -o pipefail
+set -o nounset
+set -o errexit
+
+
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+base_dir="$(cd "${script_dir}/../.."  && pwd)"
+
+boilerplate="${base_dir}"/build/codegen/header.txt
+
+
+# echo "DEBUG: script_dir: ${script_dir}"
+# echo "DEBUG: CODE_GENERATOR: ${CODE_GENERATOR}"
+# echo "DEBUG: boilerplate ${boilerplate}"
+#  echo "DEBUG: done"
+# exit 0
+
+
+
+
+
+source "${CODE_GENERATOR}/kube_codegen.sh"
+
+
 
-scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 # CODE GENERATION
 # we run deepcopy and client,lister,informer generations separately so we can use the flag "--plural-exceptions"
 # which is only known by client,lister,informer binary and not the deepcopy binary
 
 # run code deepcopy generation
-bash ${CODE_GENERATOR}/generate-groups.sh \
-    deepcopy \
-    github.com/rook/rook/pkg/client \
-    github.com/rook/rook/pkg/apis \
-    "${GROUP_VERSIONS}" \
-    --output-base "$(dirname "${BASH_SOURCE[0]}")/../../../../.." \
-    --go-header-file "${scriptdir}/boilerplate.go.txt"
+ kube::codegen::gen_helpers \
+    --boilerplate "${boilerplate}" \
+    "${base_dir}/pkg/apis" \
 
 # run code client,lister,informer generation
-bash ${CODE_GENERATOR}/generate-groups.sh \
-    client,lister,informer \
-    github.com/rook/rook/pkg/client \
-    github.com/rook/rook/pkg/apis \
-    "${GROUP_VERSIONS}" \
-    --output-base "$(dirname "${BASH_SOURCE[0]}")/../../../../.." \
-    --go-header-file "${scriptdir}/boilerplate.go.txt" \
+kube::codegen::gen_client \
+    --output-dir "${base_dir}" \
+    --output-pkg "github.com/rook/rook/pkg/client" \
+    --boilerplate "${boilerplate}" \
     --plural-exceptions "CephNFS:CephNFSes" \
+    --with-watch \
+    "${base_dir}/pkg/apis"
```

Comment:
I got it working. It seems to be that the codegen needs to be run from within the API go module. Updating the script to `cd` there in a subshell will work well

```sh
(
    # because apis is separate gomod, must change to that dir for generation
    cd ${base_dir}/pkg/apis

    # run code deepcopy generation
    kube::codegen::gen_helpers \
        --boilerplate "${boilerplate}" \
        "${base_dir}/pkg/apis"

    # run code client,lister,informer generation
    kube::codegen::gen_client \
        --output-dir "${base_dir}/pkg/client" \
        --output-pkg "github.com/rook/rook/pkg/client" \
        --boilerplate "${boilerplate}" \
        --plural-exceptions "CephNFS:CephNFSes" \
        --with-watch \
        "${base_dir}/pkg/apis"
)
```

Reply:
Fantastic, this works!

Thanks a lot, @BlaineEXE !
I updated the code and added you as co-author to the commit.